test/upgrade/prober/wathola/config/defaults.go

```diff
@@ -69,7 +72,7 @@ func defaultValues() *Config {
 		Sender: SenderConfig{
 			Address:  fmt.Sprintf("http://localhost:%v/", forwarderPort),
 			Interval: 10 * time.Millisecond,
-			Cooldown: time.Second,
+			Cooldown: 10 * time.Millisecond,
```

Comment:
The original cool down period will skew the calculation of the availability rate, keeping it same as interval will fix the problem. Do you see a problem here?

Reply:
Yes. The cooldown is to let live the service that might be overwhelmed with traffic. 

Instead of setting this default I think we should take it into account when calculating the unavailability period. That needs a little more sophisticated algorithm. Remember that this should work with whatever values of Interval, and Cooldown. Those are just default that can be changed on downstream for a reason.

Reply:
Actually 10ms is problematic anyway. My experience is that the sender cannot really send at that speed, I am only able to get around 30 ms interval in `knative-gcp` due to the delay of reply from ingress. We can of course check the real time span, but I doubt we need that level of accuracy here. As you pointed out, what matter is the unavailable period. And ideally there should be at most two unavailable period, one for upgrade and one for downgrade. I am OK with not calculating the avail rate as it does not make much sense anyway and the error detection is in the longest unavailable period, WDYT? eg. if the smoke test runs for an hour, likely the available rate will be close to 100%.

Reply:
Another supporting data point, in knative-gcp the availability is always 100%, which I believe is achievable with proper rolling update parameters.

Reply:
Yes. I think we should focus not on general availability rate, which for long test be close to 100%, but rather on unavailability periods that are longer than the specified threshold.

Say, we have got 10 unavailability periods of 2-4sec each. That should be a fail in my mind.

Something different is that rate that you mention. If we can't send an event with 10ms interval, that's a huge bug to fix in itself (to report). That's only about 100 events per second. Embarrassingly small amount. We should be able to process ~100k events per second (Visa/Mastercard level).

Reply:
The current implementation does use max unavailable period as error reporting. So we are aligned here.

I didn't mean the broker cannot handle 100 events per second, it is just the fact that the sender is single threaded and wait for response before sending the next one. So if the latency is 20 ms, the actual sending interval will be 20+10 = 30ms. Even if we set interval to be 0ms, it will still be around 20ms.

Reply:
I have changed this to 100ms, so that we can get a precision of 0.1s for the down time.